### PR TITLE
Simple traffic light state detection for simulator

### LINKED ISSUE #14: Prepare the Traffic Light Detection ROS node
- [ ] Subscribe to /vehicle/traffic_lights topic
- [ ] Publish the traffic_waypoint topic that for now should contain the TL status provided by the simulator in the /vehicle/traffic_lights topic.
- [ ] Calculate distance(s) to traffic lights ahead, maybe even more precise position if needed for classification, e.g. image coordinates.

### PR DESCRIPTION
**Pull request by mischastik**
Resolves #14 
This branch implements a very basic traffic light detection.
The result is not yet used by the waypoint_updater node.

### CODE REVIEW

File: ros/src/tl_detector/tl_detector.py [hunk 1]
```diff
@@ -68,6 +77,9 @@ def image_cb(self, msg):
             msg (Image): image from car-mounted camera
 
         """
+        if self.waypoints_tree is None:
+            return
+
         self.has_image = True
         self.camera_image = msg
         light_wp, state = self.process_traffic_lights()
```
Comment (misaksson) [inline ros/src/tl_detector/tl_detector.py:85]: I see this is not something you've changed, but I wonder if it's really okay to do image processing inside a callback? What happens if the image processing is too slow, such that another image arrives before the first one finish? Is ROS okay with callbacks not returning quickly?

File: ros/src/tl_detector/tl_detector.py [hunk 2]
```diff
@@ -49,13 +51,20 @@ def __init__(self):
...
+        self.waypoints_tree = None
+        self.n_waypoints = 0
         rospy.spin()
 
     def pose_cb(self, msg):
         self.pose = msg
 
     def waypoints_cb(self, waypoints):
         self.waypoints = waypoints
+        # Preprocess to make it faster to search for closest waypoint using the k-d tree algorithm.
```
Comment (misaksson) [inline ros/src/tl_detector/tl_detector.py:64]: There is a class WaypointSearch in waypoint_updater.py that do exactly this type of search.

**Review by misaksson — APPROVED** [review]
Looks good to me, and it seems to work properly in the simulator (viewed the /traffic_waypoint topic). I think this also resolves #14.

File: ros/src/tl_detector/tl_detector.py (hunk 1 above)
Comment (mischastik) [reply ros/src/tl_detector/tl_detector.py:85]: I guess that's fine if a queue is configured. This way it should simply drop frames if the queue is full.

File: ros/src/tl_detector/tl_detector.py (hunk 2 above)
Comment (mischastik) [reply ros/src/tl_detector/tl_detector.py:64]: Thanks, that's noted, I'll refactor that in a later step.

**Pull request merged**